### PR TITLE
Implement blue light filters

### PR DESCRIPTION
**Pull request by panicbit**
This change is [Reviewable](https://reviewable.io/reviews/aurorawright/luma3ds/1065)

### CODE REVIEW

**Comment by panicbit** [comment]
I just noticed that the LUT gets reset when waking up from standby.

File: sysmodules/rosalina/rosalina.rsf [hunk 1]
```diff
@@ -50,8 +50,12 @@ AccessControlInfo:
   # The kernel extension removes svc perms checks, so below is just to avoid a makerom error
     Backdoor: 123
     KernelSetState: 124
+    BindInterrupt: 80
```
Comment (fincs) [inline sysmodules/rosalina/rosalina.rsf]: This isn't really used, see above comment

File: sysmodules/rosalina/source/menus/screen_filters.c [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+int screenFiltersCurrentTemperature = TEMP_DEFAULT;
+
+void writeLut(u32* lut)
+{
+    u8 idx = 0;
+    do {
+        u32 pos = idx & 0xFF;
+
+        GPU_FB_TOP_COL_LUT_INDEX = pos;
```
Comment (profi200) [inline sysmodules/rosalina/source/menus/screen_filters.c]: You only need to set the index once. It increases automatically on each write.

**Review by 0pen5 — APPROVED** [review]

**Comment by therealbungus** [comment]
Three months later and this still isn't added...

**Comment by TuxSH** [comment]
Sorry for having not reviewed it yet, will do later

**Comment by lagseeing** [comment]
Did it get reviewed? I want to try this.

**Comment by 916253** [comment]
When rebased against the latest commit and built, rosalina's menu no longer opens after setting a filter. Dunno if that happened initially, but its something that needs to be resolved before this can be merged.

**Comment by panicbit** [comment]
I'll have a look at it again (but it may take a few weeks).

**Comment by TuxSH** [comment]
Sorry for the very long delay, was busy with other stuff, the code seems totally fine.

Maybe the bug is due to the fact that the menu stack size was not actually bumped (in the array declaration), will check later.

EDIT: yep it's that & I have it working now

**Pull request merged**